openmc/tallies.py

```diff
@@ -1313,10 +1196,76 @@ def get_pandas_dataframe(self, filters=True, nuclides=True,
         df['mean'] = self.mean.ravel()
         df['std. dev.'] = self.std_dev.ravel()
 
-        df.index.name = 'bin'
         df = df.dropna(axis=1)
+
+        # Expand the columns into Pandas MultiIndices for readability
+        if pd.__version__ >= '0.16':
+            columns = copy.deepcopy(df.columns.values)
+
+            # Convert all elements in columns list to tuples
+            for i, column in enumerate(columns):
+                if not isinstance(column, tuple):
+                    columns[i] = (column,)
+
+            # Make each tuple the same length
+            max_len_column = len(max(columns, key=len))
+            for i, column in enumerate(columns):
+                delta_len = max_len_column - len(column)
+                if delta_len > 0:
+                    new_column = list(column)
+                    new_column.extend(['']*delta_len)
+                    columns[i] = tuple(new_column)
+
+            # Create and set a MultiIndex for the DataFrame's columns
+            df.columns = pd.MultiIndex.from_tuples(columns)
+
         return df
 
+    def get_reshaped_data(self, value='mean'):
+        """Returns an array of tally data with one dimension per filter.
+
+        The tally data in OpenMC is stored as a 3D array with the dimensions
+        corresponding to filters, nuclides and scores. As a result, tally data
+        can be opaque for a user to directly index (i.e., without use of the
+        Tally.get_values(...) method) since one must know how to properly use
+        the number of bins and strides for each filter to index into the first
+        (filter) dimension.
+
+        This builds and returns a reshaped version of the tally data array with
+        unique dimensions corresponding to each tally filter. For example,
+        suppose this tally has arrays of data with shape (8,5,5) corresponding
+        to two filters (2 and 4 bins, respectively), five nuclides and five
+        scores. This method will return a version of the data array with the
+        with a new shape of (2,4,5,5) such that the first two dimensions
+        correspond directly to the two filters with two and four bins.
+
+        Parameters
+        ---------
+        value : str
+            A string for the type of value to return  - 'mean' (default),
+            'std_dev', 'rel_err', 'sum', or 'sum_sq' are accepted
+
+        Returns
+        -------
+        ndarray
+            The tally data array indexed by filters, nuclides and scores.
+
+        """
+
+        # Get the 3D array of data in filters, nuclides and scores
+        data = self.get_values(value=value)
+
+        # Build a new array shape with one dimension per filter
+        new_shape = ()
+        for filter in self.filters:
+            new_shape += (filter.num_bins, )
+        new_shape += (self.num_nuclides,)
+        new_shape += (self.num_score_bins,)
```

Comment:
You could make new_shape a list and then `append` each value. But, if you really feel strongly that shape should be a tuple, I understand.